File: p42586.java
import java.util.ArrayList;
public class p42586 {
    public ArrayList<Integer> solution(int[] progresses, int[] speeds) {
        ArrayList<Integer> answer = new ArrayList<>();
        int i = 0;
        while(i < progresses.length){
            int day = 0;
            int cnt = 1;
            while(progresses[i] + (speeds[i] * day) < 100) day++;
            i++;
            while(i < progresses.length && progresses[i] + (speeds[i] * day) >= 100){
                i++;
                cnt++;
            }
            answer.add(cnt);
        }
        return answer;
    }
}

File: p42586.cpp
#include<bits/stdc++.h>
using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    int i = 0;
    int day = 0;
    while(i < progresses.size()){
        int cnt = 0;
        while (day * speeds[i] + progresses[i] < 100)
            day++;
        while (day * speeds[i] + progresses[i] >= 100){
            cnt++;
            i++;
        }
        answer.push_back(cnt);
    }

    return answer;
}